### DISCUSSION TITLE
$urlskip

### DISCUSSION THREAD

**Discussion by ghajini** [uBlockOrigin/uBlock-issues#3499] [Q&A] [answered]
`https://compu-pc.com/rimworld-pc-game/`

**Ver Enlaces** has href `https://cpclinks.org/#WVVoU01HTklUVFpNZVRreVRrTTFkMWxZVGpCYVdFSnFURzAxYkdSRE9DOWthakI0VFhwbk1RPT0=`

when decoded `WVVoU01HTklUVFpNZVRreVRrTTFkMWxZVGpCYVdFSnFURzAxYkdSRE9DOWthakI0VFhwbk1RPT0=`  > `YUhSMGNITTZMeTkyTkM1d1lYTjBaWEJqTG01bGRDOC9kajB4TXpnMQ==`  > `aHR0cHM6Ly92NC5wYXN0ZXBjLm5ldC8/dj0xMzg1` > resolves finally to `https://v4.pastepc.net/?v=1385`

trouble is when i click **Ver Enlaces** ,it goes through annoying link shorteners which has some antiadb,checking click etc annoyance, if a link is sanitised it straightway goes to destination link
fix is to add `compu-pc.com##+js(acs, document.querySelectorAll, pastepc)` 

how **$urlskip** can work here when url is directly `https://cpclinks.org/#WVVoU01HTklUVFpNZVRreVRrTTFkMWxZVGpCYVdFSnFURzAxYkdSRE9DOWthakI0VFhwbk1RPT0=`
?
@gorhill ,can you see?

**Comment by gorhill** [comment]
Added support for a step to extract hash part of a URL: https://github.com/gorhill/uBlock/commit/a7aa755f1884acbf4a34e947c90129a3ddd0fa2f

**Reply by ghajini** [reply]
- i checked at `https://cpclinks.org/#WVVoU01HTklUVFpNZVRreVRrTTFkMWxZVGpCYVdFSnFURzAxYkdSRE9DOWthakI0VFhwbk1RPT0=` with `||cpclinks.org^$urlskip=# -base64` ,it does not work

- but when url is `https://cpclinks.org/#aHR0cHM6Ly92NC5wYXN0ZXBjLm5ldC8/dj0xMzg1` it work

**Reply by gorhill** [reply, marked as answer]
Try:

    ||cpclinks.org^$urlskip=# -base64 -base64 -base64

**Reply by ghajini** [reply]
nice,worked

**Reply by gorhill** [reply]
Note that I did use `https://example.com/#WVVoU01HTklUVFpNZVRreVRrTTFkMWxZVGpCYVdFSnFURzAxYkdSRE9DOWthakI0VFhwbk1RPT0= ` in my tests when adding support for `#`, and I did notice the need for multiple `-base64` directives. I am entertaining the idea of a modifier to `-base64` to force repeated calls to `-base64` until a non-base64 string is output. Maybe `-base64:n`, where `n` is the number of calls, and maybe with the special value `*` for "repeat until no more base64 string". Will think more about this.

**Reply by ghajini** [reply]
can it work with simply `-base64` 
not making much complex while internally does the job

**Reply by stephenhawk8054** [reply]
I think having a restriction `n` is better, doing automatically could cause unexpected redirections.
